### ISSUE TITLE
Bug: autofit too small on datetime columns with custom format

### ISSUE THREAD

**Issue by Christoph-AK** [jmcnamara/rust_xlsxwriter#42]
### Current behavior

![image](https://github.com/jmcnamara/rust_xlsxwriter/assets/72923608/311abec2-638d-4b0e-b2a7-a4e30f72f634)

### Expected behavior

![image](https://github.com/jmcnamara/rust_xlsxwriter/assets/72923608/1fb7d3ef-5f5d-44e7-9202-d0b16aa136a0)

### Sample code to reproduce

```rust
    let mut xls = Workbook::new();
    let sheet1 = xls.add_worksheet();
    sheet1.write(0, 0, "date")?;

    let format = Format::new().set_num_format("dd\\.mm\\.yyyy hh:mm:ss");

    let now = SystemTime::now();
    let now: DateTime<Utc> = now.into();
    let now = now.naive_local();

    sheet1.write_with_format(1, 0, &now, &format)?;

    sheet1.set_freeze_panes(1, 0)?;
    sheet1.autofit();
    xls.save(filename)?;
```

### Environment

on latest main

**Comment by jmcnamara** [comment]
This is a limitation more than a bug. `rust_xlsxwriter` doesn't currently try to apply number formatting when calculating the autofit width. A future version will try to include it in the calculation but even then it will be approximate.

The docs need to call this out more clearly so that I can at least say it is a "known" limitation.

In the meantime you should apply a fixed width based on the width of the custom format to the date columns using `worksheet.set_column_width()` or `worksheet.set_column_width_pixels()`. The `autofit()` won't override that unless the  calculated width is bigger.

**Comment by Christoph-AK** [comment]
Thanks for looking into this! 

This might be a language/region thing, but if I remember correctly the date doesn't get displayed correctly even without a custom formatting in de-de. For now I'm just setting the width to 20 manually, works ok.

Yea, living on the edge :D Love the table feature so far, was just a little bit disappointed that exisiting table header text fields get replaced with 'ColumnX' instead of getting converted to table headers automagically, but I can totally work around that :)

**Comment by jmcnamara** [comment]
> was just a little bit disappointed that exisiting table header text fields get replaced with 'ColumnX' instead of getting converted to table headers automagically

Could you explain that a bit. Did you have headers already written and then expected them to be in the table when you inserted it. Something like that?

**Comment by Christoph-AK** [comment]
Yes! Would love to be able to do 
```rust
    sheet1.write(0, 0, "header")?;
    sheet1.write(1, 0, "value1")?;
    sheet1.write(2, 0, "value2")?;

    sheet1.write(0, 1, "header2")?;
    sheet1.write(1, 1, "value3")?;
    sheet1.write(2, 1, "value4")?;

    sheet1.add_table(0, 0, 2, 1, &Table::new())?;
```

instead of 

```rust
    sheet1.write(1, 0, "value1")?;
    sheet1.write(2, 0, "value2")?;
    sheet1.write(1, 1, "value3")?;
    sheet1.write(2, 1, "value4")?;

    let mut table = Table::new();
    table.set_columns(&vec![
        TableColumn::new().set_header("header1"),
        TableColumn::new().set_header("header2")
    ]);

    sheet1.add_table(0, 0, 2, 0, &table)?;
```

**Comment by jmcnamara** [comment]
> Would love to be able to do

That is not a bad idea. I'll look into it. Not for the first table release but maybe the subsequent one.

**Comment by jmcnamara** [comment]
Closing this issue as won't fix (for now). I've improved the documentation around this feature: https://docs.rs/rust_xlsxwriter/latest/rust_xlsxwriter/struct.Worksheet.html#method.autofit

**Comment by jmcnamara** [comment]
>  was just a little bit disappointed that exisiting table header text fields get replaced with 'ColumnX' instead of getting converted to table headers automagically

 This feature is now available on the `rust_xlsxwriter` main and should be available in v0.43.0 in the next 1-2 days:

```rust
use rust_xlsxwriter::{Table, Workbook, XlsxError};

fn main() -> Result<(), XlsxError> {
    let mut workbook = Workbook::new();
    let sheet1 = workbook.add_worksheet();

    sheet1.write(0, 0, "header")?;
    sheet1.write(1, 0, "value1")?;
    sheet1.write(2, 0, "value2")?;

    sheet1.write(0, 1, "header2")?;
    sheet1.write(1, 1, "value3")?;
    sheet1.write(2, 1, "value4")?;

    sheet1.add_table(0, 0, 2, 1, &Table::new())?;

    sheet1.autofit();

    workbook.save("gh42.xlsx")?;

    Ok(())
}
```

Output:

![screenshot](https://github.com/jmcnamara/rust_xlsxwriter/assets/94267/74b72879-e288-470b-a196-da84b317dd5e)

**Comment by Christoph-AK** [comment]
Woa, brilliant. Thanks alot!

**Comment by jmcnamara** [comment]
This secondary feature request for automatic headers is now available in `rust_xlsxwriter` v0.43.0. Thanks for the prompt.
